src/rgw/CMakeLists.txt

```diff
@@ -177,7 +177,7 @@ if(WITH_RADOSGW_DBSTORE)
   list(APPEND librgw_common_srcs rgw_sal_dbstore.cc)
 endif()
 if(WITH_RADOSGW_MOTR)
-  list(APPEND librgw_common_srcs rgw_sal_motr.cc)
+  list(APPEND librgw_common_srcs rgw_sal_motr.cc motr/gc/gc.cc)
```

Comment:
We will need to add more files in future. Better to add cmakelist in motr directory and add that directory here.
ref. https://github.com/Seagate/cortx-rgw/blob/main/src/rgw/store/dbstore/CMakeLists.txt

Reply:
Yes, I tried adding a subdirectory, but the compilation failed initially, I did not try to resolve it.
Will try it again in the end.

Reply:
The below code works but I have to repeat the lines from `src/rgw/CMakeLists.txt` in the new file,
does anyone know how to avoid that.
```CMake
// src/rgw/motr/CMakeLists.txt
set(cortx_rgw_srcs
    gc/gc.cc)

add_library(cortx_rgw ${cortx_rgw_srcs})
target_include_directories(cortx_rgw PUBLIC "${CMAKE_SOURCE_DIR}/src/rgw")

// *************** repeated section start ***************
target_include_directories(cortx_rgw PRIVATE "/usr/include/motr")
target_compile_options(cortx_rgw PRIVATE "-Wno-attributes")
target_compile_definitions(cortx_rgw PRIVATE "M0_EXTERN=extern" "M0_INTERNAL=")
target_link_libraries(cortx_rgw PRIVATE motr motr-helpers)
// *************** repeated section end ***************

set(link_targets spawn)
target_link_libraries(cortx_rgw PUBLIC ${link_targets})
```

Reply:
@andriytk @siningwuseagate can you please help here?